bodono/scs-python
# switch build-time dependency declaration to PEP 517, to build against oldest version of numpy possible for each python version to avoid ABI compatibility issues.

It is best practice to build against the oldest version of Numpy a package can work with, as binaries compiled with old Numpy versions are binary compatible with newer Numpy versions, but not vice versa. Normally, this can be done by using `oldest-supported-numpy` (see https://pypi.org/project/oldest-supported-numpy/), but scs requires numpy version at least 1.7, so we must specify the versions manually for each python version.

This will fix issues like https://github.com/cvxpy/cvxpy/issues/1367 (`cvxpy` depends on `scs`).

# Test Plan
1. run `python setup.py sdist` to produce tarball.
2. `pip install cvxpy numpy==1.19.2 /code/scs-python/dist/scs-2.1.3.tar.gz`.
3. verify that cvxpy can be imported without numpy versioning issues (this failed before this change).
